src/Terrain/Maps/MapReader.java

```diff
@@ -0,0 +1,171 @@
+package Terrain.Maps;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import Engine.XYCoord;
+import Terrain.Environment;
+import Terrain.MapInfo;
+
+public class MapReader extends IMapBuilder
+{
+  private final static String mapName = "Spann Island";
```

Comment:
It looks like you copy-pasta'd and forgot to remove some things. From here through line 37 can disappear I think, plus the getMapInfo() method at the end of the file.